Tests/ApolloCodegenTests/CodeGeneration/Templates/InputObjectTemplateTests.swift

```diff
@@ -0,0 +1,312 @@
+    ])
+
+    let expected = "struct MOCKInput: InputObject {"
+
+    // when
+    let actual = InputObjectTemplate(graphqlInputObject: graphqlInputObject).render()
+
+    // then
+    expect(actual).to(equalLineByLine(expected, ignoringExtraLines: true))
+  }
+
+  func test_render_givenMixedCaseInputObjectField_generatesCorrectlyCasedSwiftDefinition() throws {
+    // given
+    let graphqlInputObject = GraphQLInputObjectType.mock("mOcK_Input", fields: [
+      GraphQLInputField.mock("field", type: .scalar(.integer()), defaultValue: nil)
+    ])
+
+    let expected = "struct MOcK_Input: InputObject {"
+
+    // when
+    let actual = InputObjectTemplate(graphqlInputObject: graphqlInputObject).render()
+
+    // then
+    expect(actual).to(equalLineByLine(expected, ignoringExtraLines: true))
+  }
+
+  // MARK: Field Type Tests
+
+  func test_render_givenSingleFieldType_generatesCorrectParameterAndInitializer_withClosingBrace() throws {
+    // given
+    let graphqlInputObject = GraphQLInputObjectType.mock("MockInput", fields: [
+      GraphQLInputField.mock("field", type: .scalar(.string()), defaultValue: nil)
+    ])
+
+    let expected = """
+      init(
+        field: GraphQLNullable<String> = nil
+      ) {
+        dict = InputDict([
+          "field": field
+        ])
+      }
+
+      var field: GraphQLNullable<String> {
+        get { dict["field"] }
+        set { dict["field"] = newValue }
+      }
+    }
+    """
+
+    // when
+    let actual = InputObjectTemplate(graphqlInputObject: graphqlInputObject).render()
+
+    // then
+    expect(actual).to(equalLineByLine(expected, atLine: 4, ignoringExtraLines: false))
+  }
+
+  func test_render_givenAllPossibleSchemaInputFieldTypes_generatesCorrectParametersAndInitializer() throws {
+    // given
+    let graphqlInputObject = GraphQLInputObjectType.mock("MockInput", fields: [
+      GraphQLInputField.mock(
+        "stringField",
+        type: .scalar(.string()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "intField",
+        type: .scalar(.integer()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "boolField",
+        type: .scalar(.boolean()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "floatField",
+        type: .scalar(.float()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "enumField",
+        type: .enum(.mock(name: "EnumValue")),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "inputField",
+        type: .inputObject(.mock(
+          "InnerInputObject",
+          fields: [
+            GraphQLInputField.mock("innerStringField", type: .scalar(.string()), defaultValue: nil)
+          ]
+        )),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "listField",
+        type: .list(.scalar(.string())),
+        defaultValue: nil
+      )
+    ])
+
+    let expected = """
+      init(
```

Comment:
Let me know your thoughts on whether we should care about styling the output to obey column widths, etc. OR simply care about the most concise output possible.

It could generate warnings for some projects if they lint code and don't tell the linter to ignore these generated files.

Reply:
> It could generate warnings for some projects if they lint code and don't tell the linter to ignore these generated files.

I don't think linters should _ever_ be run on generated files, because the code that they generate is generally set to use the standards of the project that generates them rather than the project that consumes them. 

I do think having the `init` on a separate line is worthwhile here because then you can see more clearly if there are any changes in the params of the initializer.

Reply:
> I don't think linters should ever be run on generated files

Agreed.